### ISSUE TITLE
Training baseline model doesn't converge

### ISSUE THREAD

**Issue by shauloron** [HorizonRobotics/Sparse4D#75]
Hi,
I'm trying to train the model with the provided config (R50 256x704) with code pulled on July 10, 2024. 
I'm using 4 A100 GPU's with total batch size 48.  
With the original LR 6e-4 the training diverges and grad_norm goes NaN after ~20 epochs.
When I lower the LR to 4e-4 the loss goes down and grad_norm is ok but the final model has AP=0 for all classes after 100 epochs.
I read through all the open and closed issue. Checked that the resnet50 pretrain is loading successfully. 
using the default aug_config:
data_aug_conf = {
    "resize_lim": (0.40, 0.47),
    "final_dim": input_shape[::-1],
    "bot_pct_lim": (0.0, 0.0),
    "rot_lim": (-5.4, 5.4),
    "H": 900,
    "W": 1600,
    "rand_flip": True,
    "rot3d_range": [-0.3925, 0.3925],
}

Any ideas why my training doesn't converge?

**Comment by linxuewu** [comment]
Can you obtain correct results (metrics and visualizations) by running inference with the released checkpoint?

**Comment by shauloron** [comment]
I can't seem to find any pretrained model, only pretrained resnet50 weights.
Can you please point me to the pretrained weights you released? 
Thanks

**Comment by shauloron** [comment]
ok found it now: [ckpt](https://github.com/HorizonRobotics/Sparse4D/releases/download/v3.0/sparse4dv3_r50.pth)
I'll check how it works out.

**Comment by shauloron** [comment]
I've tried using the checkpoint above and get different results than the ones in the tutorial notebook visualizations 
To start the anchor map doesn't look the same: 
![image](https://github.com/user-attachments/assets/bce86ae8-a872-4826-a73a-ff69cd068b82)
and the detection results are partial:
![image](https://github.com/user-attachments/assets/e7d35c17-809e-4172-a3dc-9eae79cc5335)

any ideas?

**Comment by shauloron** [comment]
Hi @linxuewu I would really appreciate any idea you may have on this one?

**Comment by xiao10ma** [comment]
Same as you, have you solved the problem?

For me, i didn't train and directly used the ckpt provided in the READ.md, which is

> Sparse4Dv3 | Res50 | ImageNet | 256x704.

And i replace the tutorial.ipynb's checkpoint to "ckpt/sparse4dv3_r50.pth", and finally got partial predictions:

<img width="1004" alt="Image" src="https://github.com/user-attachments/assets/f9d4c28f-0466-471c-873f-88ff9b2cf257" />

**Comment by APiaoBoDeYu** [comment]
我也有同样的问题，配置如下：
2个A100,
epoch 100,
all_batch_size=48
gpu_num=2
lr=5e-5
backbone=ResNet50
min_lr_ratio=1e-4（最终学习率为初始学习率的万分之一）
warmup_iters=500
works_per_gpu=4
但是报错：（这个错误是在20epoch结束时，进行验证tracking时报的）

```
![Image](https://github.com/user-attachments/assets/80e04363-e75d-4041-8684-7776b61b5332)

    pred_boxes = filter_eval_boxes(nusc, pred_boxes, self.cfg.class_range, verbose=verbose)                                         [637/1939]
  File "/root/miniconda3/envs/sparse/lib/python3.8/site-packages/nuscenes/eval/common/loaders.py", line 219, in filter_eval_boxes
    class_field = _get_box_class_field(eval_boxes)
  File "/root/miniconda3/envs/sparse/lib/python3.8/site-packages/nuscenes/eval/common/loaders.py", line 283, in _get_box_class_field
    raise Exception('Error: Invalid box type: %s' % box)
Exception: Error: Invalid box type: None
```
查看tensorboard，发现loss下降，但是grad_norm爆炸，我怀疑是学习率的问题，但是单纯降低学习率lr也会有这个问题。

![Image](https://github.com/user-attachments/assets/cae79b82-cea8-48a4-97a7-bf92830fdc40)
